### ISSUE TITLE
Feature request: small optimization suggestion

### ISSUE THREAD

**Issue by leon-ts** [X-Sharp/XSharpPublic#1371]
The Workarea class has a FieldIndex method that is very commonly used when reading and writing fields by name. This method itself is fast, but it still does two operations with the dictionary. But you can make one using the TryGetValue method.

**Current version:**
```
VIRTUAL METHOD FieldIndex(fieldName AS STRING) AS INT
    IF SELF:_fieldNames:ContainsKey(fieldName)
        RETURN SELF:_fieldNames[fieldName]+1
    ENDIF
    RETURN 0
```

**Version with the suggested optimization:**
```
VIRTUAL METHOD FieldIndex(fieldName AS STRING) AS INT
    IF SELF:_fieldNames:TryGetValue(fieldName, OUT fieldPos AS INT)
        RETURN fieldPos+1
    ENDIF
    RETURN 0
```

**Comment by RobertvanderHulst** [comment]
In the future: you can also fork our repository, make the change and send us a pull request. We will review the suggested change and will merge it when we find no conflicts with other code.

**Comment by RobertvanderHulst** [comment]
Included this in 2.18.0.3
